test/integration/targets/copy/tasks/main.yml

```diff
@@ -588,36 +590,38 @@
 #
 # Recursive copy of tricky symlinks
 #
-- name: Create a directory to copy from
-  file:
-    path: '{{ output_dir }}/source1'
-    state: directory
-
-- name: Create a directory outside of the tree
-  file:
-    path: '{{ output_dir }}/source2'
-    state: directory
-
-- name: Create a symlink to a directory outside of the tree
-  file:
-    path: '{{ output_dir }}/source1/link'
-    src: '{{ output_dir }}/source2'
-    state: link
-
-- name: Create a circular link back to the tree
-  file:
-    path: '{{ output_dir }}/source2/circle'
-    src: '../source1'
-    state: link
-
-- name: Create output directory
-  file:
-    path: '{{ output_dir }}/dest1'
-    state: directory
+- block:
+  - name: Create a directory to copy from
+    file:
+      path: '{{ role_path }}/tmp/source1'
```

Comment:
Updated: a temporary directory is used on the controller side (https://github.com/ansible/ansible/pull/25672/commits/e0a71a51276a528c2c013c6e4cde8d36d844ec6b#diff-cb51fd8205ec7b17fb119bd1a123ed2d).